src/mem/address_space.h

```diff
@@ -0,0 +1,204 @@
+#include "../pal/pal.h"
+#include "ds/address.h"
+#include "ds/flaglock.h"
+
+#include <array>
+#include <iostream>
+namespace snmalloc
+{
+  template<typename Pal>
+  class AddressSpaceManager : Pal
```

Comment:
Please can we have a doc comment explaining what this class is for?

It looks as if we have one, but inside the class?  I've only ever seen doc comments there in the Windows NT codebase, I don't think any open source tooling can handle them in that location.